### PR TITLE
docs: explain optimizer access in manual optimization

### LINKED ISSUE #17281: Strict warning about using optimizers manually would be needed in the docs
### 📚 Documentation

```
# Correct
self.optimizers()[0].zero_grad()
self.optimizers()[1].zero_grad()
self.manual_backward(loss)
self.optimizers()[0].step()
self.optimizers()[1].step()

# Wrong
self.optimizer0.zero_grad()
self.optimizer1.zero_grad()
self.manual_backward(loss)
self.optimizer0.step()
self.optimizer1.step()
```
I've struggled with the issue of ModelCheckpoint module not saving checkpoint files to the designated directory. And I just found out from [this issue](https://github.com/Lightning-AI/lightning/issues/13752)  that the difference of code above can cause global_step not increasing because global_step refers to the sum of all optimizer.step() calls, which leads to the situation that self._last_global_step_saved == trainer.global_step in _should_skip_saving_checkpoint() function is always True. This is very frustrating for newcomers like me because it's hardly found any relations between the coding style of accessing optimizers and global step. So I suggest global_step be more intuitive (for example, the number of training_step ends), or put some explanation in the [docs](https://lightning.ai/docs/pytorch/latest/common/optimization.html) about why it is necessary accessing optimizers with self.optimizers() function, not directly.

### PR DESCRIPTION
**Pull request by nightcityblade**
## What does this PR do?

Clarifies the manual optimization guide by explaining why optimizer steps should use the wrapper returned by `self.optimizers()`. Stepping a raw optimizer bypasses Lightning's optimizer-step tracking, which affects `Trainer.global_step`, logging, and checkpointing.

Fixes #17281

This PR introduces no breaking changes or new dependencies.

### Tests

- `uvx --from sphinx-lint sphinx-lint docs/source-pytorch/model/manual_optimization.rst`
- `uvx codespell docs/source-pytorch/model/manual_optimization.rst`
- `git diff --check upstream/master...HEAD`

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? Yes, #17281 is labeled `help wanted` and `docs`.
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? This is a documentation-only change.
- Did you write any **new necessary tests**? N/A for documentation-only changes.
- [x] Did you verify new and **existing tests pass** locally with your changes? The focused documentation checks listed above pass.
- Did you list all the **breaking changes** introduced by this pull request? N/A; there are no breaking changes.
- Did you **update the CHANGELOG**? N/A for documentation-only changes.

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Comment by nightcityblade** [comment]
CI note: I checked the failed Read the Docs build. Sphinx aborts while loading the external intersphinx inventory at `https://lightning.ai/docs/pytorch/stable/objects.inv`; that URL returned HTML (`invalid inventory header: <!doctype html>`). The failure occurs before rendering this PR's changed page and is unrelated to the added warning. All repository-hosted CI jobs and the focused RST checks pass, so no source change is warranted.

**Comment by nightcityblade** [comment]
CI note: I inspected the Read the Docs failure. The build stops while loading intersphinx inventories because https://lightning.ai/docs/pytorch/stable/objects.inv returned HTML instead of a valid inventory; it does not report an error in the changed manual-optimization guide. All other checks and the focused documentation validation pass. Could a maintainer please rerun the Read the Docs build once the external inventory is available?
